scvi/data/fields/_matrix_field.py

```diff
@@ -0,0 +1,660 @@
+import logging
+import warnings
+from typing import Dict, List, Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+import rich
+from anndata import AnnData
+from pandas.api.types import CategoricalDtype
+
+from scvi.data import _constants
+from scvi.data._utils import (
+    _check_nonnegative_integers,
+    _make_column_categorical,
+    _verify_and_correct_data_format,
+)
+
+from ._base_field import BaseAnnDataField
+from ._mudata import MuDataWrapper
+
+logger = logging.getLogger(__name__)
+
+
+class BaseMatrixField(BaseAnnDataField):
```

Comment:
```suggestion
class BaseArrayLikeField(BaseAnnDataField):
```

Thoughts?

Reply:
Sounds reasonable, although the reason it was named BaseMatrixField was because I thought varm and obsm both stand for matrix.

Reply:
ah I see, I'm not sure. But when I think of matrix I think of np matrix, where arraylike better captures the data structures

Reply:
Fair enough, I think I'm fine with either so I'll go with your suggestion.